code/game/objects/items.dm

```diff
@@ -1355,3 +1355,7 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 	user.swap_hand()
 	user.update_inv_l_hand(0)
 	user.update_inv_r_hand()
+
+/obj/item/proc/on_vend(faction)
+	if(usr.put_in_any_hand_if_possible(src, warning = FALSE))
```

Comment:
Never use usr, we do not allow it on this codebase for new code :)

Reply:
Removed all uses of usr in on_vend. I am still using usr in the vendor code (new_marine_vendors.dm) because the ui_act already makes extensive use of it, and I don't think I can remove any of it without refactors outside the scope of this PR.